Classes/Event/EventWithMetadata.php

```diff
@@ -0,0 +1,56 @@
+<?php
+namespace Neos\EventSourcing\Event;
+
+/*
+ * This file is part of the Neos.EventSourcing package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+/**
+ * Event wrapper which provides metadata additional to the event
+ */
+final class EventWithMetadata implements EventInterface
+{
+    /**
+     * @var EventInterface
+     */
+    private $event;
+
+    /**
+     * @var array
+     */
+    private $metadata;
+
+    /**
+     * EventWithMetadata constructor.
```

Comment:
&lt;nitpick&gt;This comment doesn't really add any value, does it?&lt;/nitpick&gt;

Reply:
Agreed. But our current coding guidelines / other code we have in the Neos project suggests that you always have a doc block for a method.

We discussed that in Kiel last week. I'm all in favor of changing these CGLs, but let's do that first and then make sure everything is consistent. My point is only consistency, not beauty.

Reply:
Right, this makes sense!